crates/frontend/src/components/type_template_form.rs

```diff
@@ -0,0 +1,124 @@
+pub mod utils;
+
+use crate::components::type_template_form::utils::create_type;
+use crate::components::{button::Button, type_template_form::utils::update_type};
+use leptos::*;
+use serde_json::{from_str, json, Value};
+use web_sys::MouseEvent;
+
+#[component]
+pub fn type_template_form<NF>(
+    #[prop(default = false)] edit: bool,
+    #[prop(default = String::new())] type_name: String,
```

Comment:
should we have default type_name also if we have default schema

Reply:
Hmm risky, that shouldn't be the case